### PR TITLE
fix: support multiple type depending on the reponse code

### PR DESCRIPTION
**Pull request by cpaulve-1A**
## Related issues

- :bug: Fixes #(issue)
- :rocket: Feature #(issue)

### CODE REVIEW

File: packages/@ama-sdk/core/src/fwk/core/api-client.ts [hunk 1]
```diff
@@ -51,8 +51,8 @@ export interface ApiClient {
   processFormData(data: any, type: string): FormData | string;
 
   /** Process HTTP call */
-  processCall<T>(url: string, options: RequestOptions, apiType: ApiTypes | string, apiName: string, reviver?: ReviverType<T> | undefined, operationId?: string): Promise<T>;
-
+  processCall<T>(url: string, options: RequestOptions, apiType: ApiTypes | string, apiName: string, reviver?: ReviverType<T> | undefined |
+    {[key: number]: ReviverType<T> | undefined}, operationId?: string): Promise<T>;
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/core/api-client.ts:55]: `{[key: number]: ReviverType<T> | undefined}` or `Record<number, ReviverType<T> | undefined>` ?

File: packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts [hunk 2]
```diff
@@ -150,19 +153,34 @@ export class ApiFetchClient implements ApiClient {
...
+          operationId,
+          url,
+          origin
+        })) : [];
+    } else if (!!response?.status && reviver[response.status]) {
+      replyPlugins = this.options.replyPlugins ?
+        this.options.replyPlugins.map((plugin) => plugin.load<T>({
+          dictionaries: root && root.dictionaries,
+          response,
+          reviver: reviver[response!.status],
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts:175]: would not be simpler to have a ternary here?

File: packages/@ama-sdk/core/src/fwk/core/api-client.ts (hunk 1 above)
Comment (matthieu-crouzet) [reply packages/@ama-sdk/core/src/fwk/core/api-client.ts:55]: `{[responseCode: number]: ReviverType<T> | undefined}`
so it easier to understand what should be the key

**Review by matthieu-crouzet — APPROVED** [review]

File: packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts [hunk 3]
```diff
@@ -141,21 +143,20 @@ export class ApiFetchClient implements ApiClient {
...
-      exception = new ResponseJSONParseError(e.message || 'Fail to parse response body', response && response.status || 0, body, { apiName, operationId, url, origin });
+      exception = new ResponseJSONParseError(e.message || 'Fail to parse response body', response && response.status || 0, body, {apiName, operationId, url, origin});
     }
-
     const replyPlugins = this.options.replyPlugins ?
       this.options.replyPlugins.map((plugin) => plugin.load<T>({
         dictionaries: root && root.dictionaries,
         response,
-        reviver,
+        reviver: typeof reviver === 'function' || typeof reviver === 'undefined' ? reviver : (response?.status && reviver[response.status]) || undefined,
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts:159]: not sure the `|| undefined1 bring something, but OK :D

File: packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts [hunk 4]
```diff
@@ -164,6 +165,7 @@ export class ApiFetchClient implements ApiClient {
         origin
       })) : [];
 
+
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts:168]: additional empty line? :D

**Review by kpanot — APPROVED** [review]

**Review by matthieu-crouzet — DISMISSED** [review]

File: packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts (hunk 3 above)
Comment (cpaulve-1A) [reply packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts:159]: image

File: packages/@ama-sdk/schematics/schematics/resources/6.6.0.jar
Comment (mrednic-1A) [inline packages/@ama-sdk/schematics/schematics/resources/6.6.0.jar:1]: is this supposed to be versioned ? :D

Comment (cpaulve-1A) [reply packages/@ama-sdk/schematics/schematics/resources/6.6.0.jar:1]: not really 👍

**Review by matthieu-crouzet — APPROVED** [review]

**Review by mrednic-1A — APPROVED** [review]

File: packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts (hunk 3 above)
Comment (kpanot) [reply packages/@ama-sdk/core/src/fwk/clients/api-fetch-client.ts]: I missed the `0` indead. TS is good !

**Pull request merged**